src/complex/Common/Bit.hpp

```diff
@@ -91,4 +79,40 @@ To bit_cast(const From& src) noexcept
   std::memcpy(&dst, &src, sizeof(To));
   return dst;
 }
+
+template <class T>
+inline constexpr T byteswap(T value) noexcept
+{
+  static_assert(std::is_arithmetic_v<T>, "byteSwap only works on arithmetic types");
+  if constexpr(sizeof(T) == sizeof(uint8))
+  {
+    return value;
+  }
+  else if constexpr(sizeof(T) == sizeof(uint16))
+  {
+    return ((value >> 8) & 0xFFu) | ((value & 0xFFu) << 8);
```

Comment:
I'd maybe move this into a `byteswap16` function to be consistent with the other two.